x/interchainstaking/keeper/callbacks_test.go

```diff
@@ -1607,5 +1611,119 @@ func (suite *KeeperTestSuite) TestDelegationAccountBalanceCallback() {
 	})
 }
 
+func decodeBase64NoErr(str string) []byte {
+	decoded, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		fmt.Println("Error decoding: ", str)
+		panic(err)
+	}
+	return decoded
+}
+
+func (suite *KeeperTestSuite) TestDepositTxCallback() {
+	setupIbc := func() (*app.Quicksilver, sdk.Context) {
+		// reset test suite
+		suite.SetupTest()
+		suite.setupTestZones()
+
+		// setup quicksilver test app
+		quicksilver := suite.GetQuicksilverApp(suite.chainA)
+		quicksilver.InterchainstakingKeeper.CallbackHandler().RegisterCallbacks()
+
+		// get chainA context
+		ctx := suite.chainA.GetContext()
+
+		// get zone chainB context
+		zone, _ := quicksilver.InterchainstakingKeeper.GetZone(ctx, suite.chainB.ChainID)
+		zone.DepositAddress.IncrementBalanceWaitgroup()
+		zone.WithdrawalAddress.IncrementBalanceWaitgroup()
+		quicksilver.InterchainstakingKeeper.SetZone(ctx, &zone)
+
+		// get the tx from fixture
+		txWithProofBz := decodeBase64NoErr(localDepositTxFixture)
+		txRes := icqtypes.GetTxWithProofResponse{}
+		err := quicksilver.InterchainQueryKeeper.IBCKeeper.Codec().Unmarshal(txWithProofBz, &txRes)
+		suite.NoError(err)
+
+		// update payload header to ensure we can validate it.
+		txRes.Header.Header.Time = ctx.BlockTime()
+		// setup ClientConsensusState for checking Header validation
+		// Cheat, and set the client state and consensus state for 07-tendermint-0 to match the incoming header.
+		quicksilver.IBCKeeper.ClientKeeper.SetClientState(ctx, "07-tendermint-0", lightclienttypes.NewClientState("gaiatest-1", lightclienttypes.DefaultTrustLevel, time.Hour, time.Hour, time.Second*50, txRes.Header.TrustedHeight, []*ics23.ProofSpec{}, []string{}, false, false))
+		quicksilver.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, "07-tendermint-0", txRes.Header.TrustedHeight, txRes.Header.ConsensusState())
+
+		return quicksilver, ctx
+	}
+
+	testCases := []struct {
+		name          string
+		txHash        string
+		txWithProofbz []byte
+		chainID       string
+		expectErr     bool
+	}{
+		{
+			"Deposit transaction successful",
+			// txHash that come from DepositTxFixture
+			"b1f1852d322328f6b8d8cacd180df2b1cbbd3dd64536c9ecbf1c896a15f6217a",
+			decodeBase64NoErr(localDepositTxFixture),
+			suite.chainB.ChainID,
+			false,
+		},
```

Comment:
Just need to add the deterministic for each field like my suggestion then we can merge, thank @hoangdv2429 
```suggestion
				{
			name:          "Deposit transaction successful",
			// txHash that come from DepositTxFixture
			txHash:        "b1f1852d322328f6b8d8cacd180df2b1cbbd3dd64536c9ecbf1c896a15f6217a",
			txWithProofbz: decodeBase64NoErr(localDepositTxFixture),
			chainID:       suite.chainB.ChainID,
			expectErr:     false,
		},
```